### PR TITLE
AdminAbility: fixing admin namespace permissions

### PR DESCRIPTION
**Pull request by davidwessman**
Adding AdminAbility for custom permissions
Adding a Admin::BaseController which sets current_ability to AdminAbility
Changing all admin-namespaced controllers to inherit from said BaseController

### CODE REVIEW

File: app/controllers/admin/rents_controller.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def update
   end
 
   def destroy
-    @rent.destroy
+    @rent.destroy!
```
Comment (jforberg) [inline app/controllers/admin/rents_controller.rb]: Now you see why using the ! method is important :) it's so easy to miss and then you have a silent error.

**Comment by jforberg** [comment]
This looks like a reasonably clean solution. Good work. +1

**Comment by navro** [comment]
You need to change a lot of `can?` to `can_administrate?` in the views before merging this

**Comment by davidwessman** [comment]
Yep - I did almost all of them this morning - didn't push.

**Comment by davidwessman** [comment]
I got an update for Faq ready to merge after this one.
Right now the Faq will not function properly (and is not used either)

Does it look okay otherwise?

**Comment by navro** [comment]
+1
Great solution :smile:

**Pull request merged**